### PR TITLE
[stable9] Use the correct realm for basic authentication - fixes #23427

### LINKED ISSUE #23427: OC9 - Calendar authentication credentials need to be re-entered because the calendar naming has changed
## Expected behavior
After upgrading from OC8.2.2 to OC9.0 it should be possible to access and use all
CalDAV calendars without re-entering the authentication credentials.

## Current behavior
The authentication credentials for all subscribed calendars need to be re-entered
in e.g. Thunderbird/Lightning because the calendar naming has slightly been changed.

Example:
OC 8.x: https://your-server-url (**ownCloud** (Name of calendar))
OC 9.x: https://your-server-url (**sabre/dav** (Name of calendar)) 

#### Server Configuration
OS: Linux 3.2.75
Web server: Apache2 2.4.18
Database: MariaDB 5.5.46
PHP version: 5.6.17
ownCloud version: 9.0.0
Calendar app version: 1.0.0 (20160320)
Updated from OC8.2.2

#### Client Configuration
Browser: Thunderbird 38.7.0/Lightning 4.0.7
Operating system: Windows 7

### PR DESCRIPTION
**Pull request by PVince81**
Backport of https://github.com/owncloud/core/pull/25046 to stable9.

Please note that I had to remove one block from the conflict because we don't hate the "X-Requested-With" logic at all in OC 9.0.

Please review @DeepDiver1975 @SergioBertolinSG 

I've cherry-picked the automated test from https://github.com/owncloud/core/pull/26712 here

### CODE REVIEW

**Comment by PVince81** [comment]
Fixes https://github.com/owncloud/core/issues/26720

**Comment by PVince81** [comment]
I retested fed shares in both directions and they still work.

**Comment by PVince81** [comment]
Boo...

```
error: index-pack died of signal 15
11:07:42 fatal: index-pack failed
```

**Comment by PVince81** [comment]
@DeepDiver1975 review ? It's your fix that I backported

File: apps/dav/lib/connector/publicauth.php [hunk 1]
```diff
@@ -26,6 +26,13 @@
 
 namespace OCA\DAV\Connector;
 
+use Sabre\DAV\Auth\Backend\AbstractBasic;
```
Comment (DeepDiver1975) [inline apps/dav/lib/connector/publicauth.php]: unused - see 36

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**